waiter/test/waiter/reporters_test.clj

```diff
@@ -0,0 +1,173 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns waiter.reporters-test
+  (:require [clj-time.core :as t]
+            [clojure.test :refer :all]
+            [metrics.core :as mc]
+            [metrics.counters :as counters]
+            [waiter.metrics :refer :all]
+            [waiter.reporter :as r])
+  (:import (clojure.lang ExceptionInfo)
+           (com.codahale.metrics MetricFilter MetricRegistry ConsoleReporter)
+           (com.codahale.metrics.graphite GraphiteReporter GraphiteSender)
+           (java.io PrintStream ByteArrayOutputStream)))
+
+(def ^:private all-metrics-match-filter (reify MetricFilter (matches [_ _ _] true)))
+
+(defmacro with-isolated-registry
+  [& body]
+  `(with-redefs [mc/default-registry (MetricRegistry.)]
+     (.removeMatching mc/default-registry all-metrics-match-filter)
+     (do ~@body)
+     (.removeMatching mc/default-registry all-metrics-match-filter)))
+
+(deftest console-reporter-bad-schema
+  (is (thrown-with-msg? ExceptionInfo #"period-ms missing-required-key"
+                        (r/validate-console-reporter-config {:extra-key 444}))))
+(deftest console-reporter-good-schema
```

Comment:
We should have new lines between `deftests`. Alternatively, you can have one tests and place all of these inside individual `testing` clauses.